feat: add FunFair.CodeAnalysis to owned packages registry (#931)
credfeto/cs-template: ai/global/dotnet.instructions.md

## Test Dependencies

> The `FunFair.Test.*` packages in this section are owned by the organisation — see [dotnet-owned-packages.instructions.md](dotnet-owned-packages.instructions.md) for the GitHub source repository.

Reviewer: **Finding (CONFIRMED):** `"in this section"` is inaccurate — `FunFair.Test.Infrastructure` is not documented in the Test Dependencies section.

The attribution at line 212 reads: _"The `FunFair.Test.*` packages **in this section** are owned by the organisation…"_

But `FunFair.Test.Infrastructure` appears only in the **"FunFair.Test.* — Prefer Library Code Over Custom Implementations"** section (line 251), not here. An agent reading this attribution and then scanning the Test Dependencies section (lines 214–235) will find `FunFair.Test.Common` and `FunFair.Test.Source.Generator` — but not `FunFair.Test.Infrastructure` — and may incorrectly conclude that `FunFair.Test.Infrastructure` is not an org-owned package.

**Suggested fix:** Change "in this section" to "in this file" (or drop the qualifier entirely), since the wildcard `FunFair.Test.*` is meant to cover all three packages regardless of where they appear:

```suggestion
The `FunFair.Test.*` packages in this file are owned by the organisation — see [dotnet-owned-packages.instructions.md](dotnet-owned-packages.instructions.md) for the GitHub source repository.
```

Reviewer: Fixed in 79e87483 — changed 'in this section' to 'in this file' so the ownership attribution covers FunFair.Test.Infrastructure regardless of which section it appears in.

## FunFair.Test.* — Prefer Library Code Over Custom Implementations (MANDATORY)

> The `FunFair.Test.*` packages in this section are owned by the organisation — see [dotnet-owned-packages.instructions.md](dotnet-owned-packages.instructions.md) for the GitHub source repository.

Reviewer: Duplicate ownership attribution sentence — identical to the one already added at line 210 in the `## Test Dependencies` section. These two sections are adjacent in the same file; placing word-for-word identical notes in both violates the index.md no-duplication rule. The second instance should be removed (agents reading the file will already have seen the note from the Test Dependencies section) or reworded to be section-specific (e.g. 'This section discusses what these owned packages provide rather than which ones to reference — see the Test Dependencies section above for how to add them.').

Reviewer: Fixed in 3806a853 — removed the duplicate ownership attribution sentence from the FunFair.Test.* — Prefer Library Code section. The note at line 210 in Test Dependencies is sufficient since the two sections are adjacent in the same file.

## FunFair.Test.* — Prefer Library Code Over Custom Implementations (MANDATORY)

> The `FunFair.Test.*` packages in this section are org-owned — see [dotnet-owned-packages.instructions.md](dotnet-owned-packages.instructions.md) for their GitHub source repositories.

Reviewer: **Finding (CONFIRMED):** \"their GitHub source repositories\" uses the wrong plural — all three `FunFair.Test.*` packages share a single repository.

The `dotnet-owned-packages.instructions.md` registry consolidates `FunFair.Test.Common`, `FunFair.Test.Infrastructure`, and `FunFair.Test.Source.Generator` onto a **single row** pointing to one repo (`funfair-server-test`). Saying \"repositories\" (plural) implies multiple repos exist, which could send an agent searching for a second repository that does not exist.

Additionally, this sentence says \"org-owned\" while the equivalent sentence at line 212 says \"owned by the organisation\" — two different phrasings for the same ownership concept in the same file.

```suggestion
The `FunFair.Test.*` packages in this section are owned by the organisation — see [dotnet-owned-packages.instructions.md](dotnet-owned-packages.instructions.md) for the GitHub source repository.
```

Reviewer: Fixed in b582f046 — changed 'repositories' to 'repository' (all three FunFair.Test.* packages share the single funfair-server-test repo) and aligned phrasing to 'owned by the organisation' to match line 212.

> ## String Comparison

Reviewer: Fixed in 5870d027 — changed 'source repository' to 'GitHub source repository' to eliminate the ambiguity with NuGet package sources/feeds.

## String Comparison

> FFS rules in this section are enforced by the `FunFair.CodeAnalysis` package — see [dotnet-owned-packages.instructions.md](dotnet-owned-packages.instructions.md) for the GitHub source repository.

Reviewer: **Finding (PLAUSIBLE):** Attribution sentence for `FunFair.CodeAnalysis` does not state org ownership, unlike the three parallel attributions in this PR.

The sentence reads: _"FFS rules in this section are enforced by the `FunFair.CodeAnalysis` package — see dotnet-owned-packages.instructions.md…"_

The other three new attributions use "is an org-owned package" or "are owned by the organisation". This one only says the package *enforces* rules, without stating it is org-owned. An agent reading only this sentence would not know to read the source rather than decompile.

(The link destination does make ownership clear, so impact is low — but consistency aids pattern matching.)

```suggestion
`FunFair.CodeAnalysis` is an org-owned package that enforces FFS rules in this section — see [dotnet-owned-packages.instructions.md](dotnet-owned-packages.instructions.md) for the GitHub source repository.
```

Reviewer: Fixed in ed2d71b4 — rephrased to 'FunFair.CodeAnalysis is an org-owned package that enforces FFS rules in this section' so it is consistent with the other three attribution sentences.
